File: task-03/Elixir/SubTask_4.ex
case File.read("input.txt") do
  {:ok,content} ->
    num=String.to_integer(content)
    n=if rem(num,2)==0 do
      num |> div(2)
    else
      (num+1) |> div(2)
    end
    upper=for i<-1..n do
      String.duplicate(" ",n-i) <> String.duplicate("*",2*i-1) <> "\n"
    end
    lower=for i<-n-1..1 do
      String.duplicate(" ",n-i) <> String.duplicate("*",2*i-1) <> "\n"
    end
    diamond=Enum.join(upper++lower)
    case File.write("output.txt",diamond) do
      :ok ->
        IO.puts "copied to output.txt"
      {:error, reason} ->
        IO.puts "#{reason}"
    end
  {:error, reason} ->
    IO.puts "#{reason}"
end

File: task-03/Haskell/SubTask_4.hs
import System.IO
import Control.Monad (forM_)

main::IO ()
main=do
    input<-readFile "input.txt"
    let num=(read input::Int)
    let n=ceiling(fromIntegral num/2.0)
    forM_ [1..n] $ \i -> do
        forM_ [1..(n-i)] $ \_ -> do
            appendFile "output.txt" " "
        forM_ [1..(2*i-1)] $ \_ -> do
            appendFile "output.txt" "*"
        appendFile "output.txt" "\n"
    forM_ (reverse [1..(n-1)]) $ \i -> do
        forM_ [1..(n-i)] $ \_ -> do
            appendFile "output.txt" " "
        forM_ [1..(2*i-1)] $ \_ -> do
            appendFile "output.txt" "*"
        appendFile "output.txt" "\n"